modules/core/04-channel/keeper/upgrade.go

```diff
@@ -261,76 +199,69 @@ func (k Keeper) WriteUpgradeTryChannel(
-	}
+			portKeeper, ok := k.portKeeper.(*portkeeper.Keeper)
+			if !ok {
+				panic("todo: handle this situation")
+			}
 
-	cbs, found := portKeeper.Router.GetRoute(module)
-	if !found {
-		return errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module)
-	}
+			cbs, found := portKeeper.Router.GetRoute(module)
+			if !found {
+				return errorsmod.Wrapf(porttypes.ErrInvalidRoute, "route not found to module: %s", module)
+			}
 
-	cbs.OnChanUpgradeRestore(ctx, portID, channelID)
+			cbs.OnChanUpgradeRestore(ctx, portID, channelID)
+	*/
 	return nil
 }
 
-// getUpgradeTryConnectionEnd returns the connection end that should be used. During crossing hellos, the restore
-// channel connection end is used, while in a regular flow the current channel connection end is used.
-func (k Keeper) getUpgradeTryConnectionEnd(ctx sdk.Context, portID string, channelID string, currentChannel types.Channel) (exported.ConnectionI, error) {
-	isCrossingHellos := currentChannel.State == types.INITUPGRADE
-	if isCrossingHellos {
-		// fetch restore channel
-		restoreChannel, found := k.GetUpgradeRestoreChannel(ctx, portID, channelID)
-		if !found {
-			return nil, errorsmod.Wrapf(types.ErrChannelNotFound, "port ID (%s) channel ID (%s)", portID, channelID)
-		}
-		connectionEnd, err := k.GetConnection(ctx, restoreChannel.ConnectionHops[0])
-		if err != nil {
-			return nil, err
-		}
-		return connectionEnd, nil
+// validateProposedUpgradeFields validates the proposed upgrade fields against the existing channel.
+// It returns an error if the following constraints are not met:
+// - there exists at least one valid proposed change to the existing channel fields
+// - the proposed order is a subset of the existing order
+// - the proposed connection hops do not exist
+// - the proposed version is non empty (checked in ModifiableUpgradeFields.ValidateBasic())
+func (k Keeper) validateProposedUpgradeFields(ctx sdk.Context, existingChannel types.Channel, proposedUpgrade types.ModifiableUpgradeFields) error {
+	currentFields := types.ModifiableUpgradeFields{
+		Ordering:       existingChannel.Ordering,
+		ConnectionHops: existingChannel.ConnectionHops,
+		Version:        existingChannel.Version,
+	}
+	if reflect.DeepEqual(proposedUpgrade, currentFields) {
+		return errorsmod.Wrap(types.ErrChannelExists, "existing channel end is identical to proposed upgrade channel end")
 	}
 
-	// use current channel
-	connectionEnd, err := k.GetConnection(ctx, currentChannel.ConnectionHops[0])
-	if err != nil {
-		return nil, err
+	if !currentFields.Ordering.SubsetOf(proposedUpgrade.Ordering) {
+		return errorsmod.Wrap(types.ErrInvalidChannelOrdering, "channel ordering must be a subset of the new ordering")
 	}
-	return connectionEnd, nil
+
+	if !k.connectionKeeper.HasConnection(ctx, proposedUpgrade.ConnectionHops[0]) {
+		return errorsmod.Wrapf(connectiontypes.ErrConnectionNotFound, "failed to retrieve connection: %s", proposedUpgrade.ConnectionHops[0])
+	}
```

Comment:
This connection should also be open right? It existing isn't enough for this to be valid.